aten/src/ATen/core/CustomTensorImpl.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <ATen/core/stack.h>
+#include <ATen/core/TensorBody.h>
+#include <c10/core/TensorImpl.h>
+
+using Impl = void(const c10::FunctionSchema& schema, torch::jit::Stack*);
+
+namespace at {
+
+class CAFFE2_API CustomTensorImpl : public TensorImpl {
+  size_t tag_;
+  std::shared_ptr<void> storage_;
```

Comment:
Let's not call it a storage, it's very confusing with our native notion of storage (i.e., it doesn't have anything to do with storage!)

Reply:
it's where the user defined tensor's data etc is actually stored, but I agree its confusing with Storage already a reserved name in the codebase